src/bgw_policy/drop_chunks.c

```diff
@@ -23,12 +23,52 @@ static ScanTupleResult
 bgw_policy_drop_chunks_tuple_found(TupleInfo *ti, void *const data)
 {
 	BgwPolicyDropChunks **policy = data;
-
-	*policy = STRUCT_FROM_TUPLE(ti->tuple,
-								ti->mctx,
-								BgwPolicyDropChunks,
-								FormData_bgw_policy_drop_chunks);
-
+	bool nulls[Natts_bgw_policy_drop_chunks];
+	Datum values[Natts_bgw_policy_drop_chunks];
+	/* cannot use STRUCT_FROM_TUPLE because some fields in the result may be NULLs*/
+	heap_deform_tuple(ti->tuple, ti->desc, values, nulls);
+
+	*policy = MemoryContextAllocZero(ti->mctx, sizeof(BgwPolicyDropChunks));
+	Assert(!nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_job_id)]);
+	(*policy)->fd.job_id =
+		DatumGetInt32(values[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_job_id)]);
+
+	Assert(!nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_hypertable_id)]);
+	(*policy)->fd.hypertable_id =
+		DatumGetInt32(values[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_hypertable_id)]);
+
+	Assert(!nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_interval_support)]);
+	(*policy)->fd.interval_support =
+		DatumGetBool(values[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_interval_support)]);
+
+	if ((*policy)->fd.interval_support)
+	{
+		Assert(!nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_older_than_interval)]);
+		Assert(nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_older_than_integer)]);
+
+		(*policy)->fd.older_than_interval = *DatumGetIntervalP(
+			values[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_older_than_interval)]);
+	}
+	else
+	{
+		Assert(nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_older_than_interval)]);
+		Assert(!nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_older_than_integer)]);
+
+		(*policy)->fd.older_than_integer = DatumGetInt64(
+			values[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_older_than_integer)]);
+	}
+
+	Assert(!nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_cascade)]);
+	(*policy)->fd.cascade =
+		DatumGetBool(values[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_cascade)]);
+
+	Assert(
+		!nulls[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_cascade_to_materializations)]);
+	(*policy)->fd.cascade_to_materializations = DatumGetBool(
+		values[AttrNumberGetAttrOffset(Anum_bgw_policy_drop_chunks_cascade_to_materializations)]);
+
+	// Q:: why scan continue? Is it not wrong to have multiple? And even if there are multiple, this
+	// would just take the last one?
 	return SCAN_CONTINUE;
```

Comment:
Should this be SCAN_DONE?

Reply:
Its scan continue so that the calling function can check the count and error if > 1. Can you verify this check happens.

Reply:
I see, verified, it is being checked

Reply:
DISCUSS: I later noticed that if I leave older_than as deformed struct inside the `FormData_bgw_policy_drop_chunks` instead of a Datum, then I cannot just use STRUCT_FROM_TUPLE.

The reason to leave older_than deformed is to make the later code using older_than simpler.